evm/src/fuzz/invariant/executor.rs

```diff
@@ -0,0 +1,290 @@
+                    collect_created_contracts(
+                        &state_changeset,
+                        self.project_contracts,
+                        self.setup_contracts,
+                        targeted_contracts.clone(),
+                        &mut created_contracts,
+                    );
+
+                    // Commit changes to the database.
+                    executor.borrow_mut().backend_mut().db.commit(state_changeset);
+
+                    fuzz_runs.push(FuzzCase { calldata: calldata.clone(), gas, stipend });
+
+                    if !reverted {
+                        if assert_invariants(
+                            test_contract_abi,
+                            &executor.borrow(),
+                            invariant_address,
+                            &invariants,
+                            &inputs,
+                            &mut failures.borrow_mut(),
+                        )
+                        .is_err()
+                        {
+                            break 'fuzz_run
+                        }
+                    } else {
+                        failures.borrow_mut().reverts += 1;
+
+                        // The user might want to stop all execution if a revert happens to
+                        // better bound their testing space.
+                        if test_options.fail_on_revert {
+                            let error = InvariantFuzzError::new(
+                                invariant_address,
+                                None,
+                                test_contract_abi,
+                                &result,
+                                &inputs,
+                                &[],
+                            );
+
+                            failures.borrow_mut().revert_reason = Some(error.revert_reason.clone());
+
+                            // Hacky to provide the full error to the user.
+                            for invariant in invariants.iter() {
+                                failures
+                                    .borrow_mut()
+                                    .failed_invariants
+                                    .insert(invariant.name.clone(), Some(error.clone()));
+                            }
+
+                            break 'fuzz_run
+                        }
+                    }
+
+                    // Generates the next call from the run using the recently updated
+                    // dictionary.
+                    inputs.extend(
+                        strat
+                            .new_tree(&mut branch_runner.borrow_mut())
+                            .map_err(|_| TestCaseError::Fail("Could not generate case".into()))?
+                            .current(),
+                    );
+                }
+
+                // Before each run, we must reset the database state.
+                executor.borrow_mut().backend_mut().db = clean_db.clone();
```

Comment:
this won't work anymore with https://github.com/foundry-rs/foundry/pull/2301

instead the Backend should be cloned

why does this need mutable access and use the same as the standard fuzzer that calls non committing?